java/vector/src/main/java/org/apache/arrow/vector/file/json/JsonFileWriter.java

```diff
@@ -85,22 +92,61 @@ public JsonFileWriter(File outputFile, JSONWriteConfig config) throws IOExceptio
     }
   }
 
-  public void start(Schema schema) throws IOException {
-    this.schema = schema;
+  public void start(Schema schema, DictionaryProvider provider) throws IOException {
+    List<Field> fields = new ArrayList<>(schema.getFields().size());
+    Set<Long> dictionaryIdsUsed = new HashSet<>();
+    this.schema = schema;  // Store original Schema to ensure batches written match
+
+    // Convert fields with dictionaries to have dictionary type
+    for (Field field: schema.getFields()) {
+      fields.add(DictionaryUtility.toMessageFormat(field, provider, dictionaryIdsUsed));
+    }
+    Schema updatedSchema = new Schema(fields, schema.getCustomMetadata());
```

Comment:
aren't the dictionary-encoded fields already encapsulated in the schema? Or do you mean the dictionary batches?